chainstate/tx-verifier/src/transaction_verifier/error.rs

```diff
@@ -80,12 +81,22 @@ pub enum ConnectTransactionError {
     TxIndexError(#[from] TxIndexError),
     #[error("Error from TransactionVerifierStorage: {0}")]
     TransactionVerifierError(#[from] TransactionVerifierStorageError),
-    #[error("BlockUndo error: {0}")]
-    BlockUndoError(#[from] BlockUndoError),
+    #[error("utxo BlockUndo error: {0}")]
+    UtxoBlockUndoError(#[from] utxo::UtxosBlockUndoError),
+    #[error("accounting BlockUndo error: {0}")]
+    AccountingBlockUndoError(#[from] pos_accounting::AccountingBlockUndoError),
     #[error("Failed to sum amounts of burns in transaction: {0}")]
     BurnAmountSumError(Id<Transaction>),
     #[error("Attempt to spend burned amount in transaction")]
     AttemptToSpendBurnedAmount,
+    #[error("PoS accounting error")]
+    PoSAccountingError(#[from] pos_accounting::Error),
+    #[error("PoS accounting undo is missing for transaction {0}")]
+    MissingPoSAccountingUndo(Id<Transaction>),
+    #[error("No token inputs is allowed for PoS accounting operations {0}")]
+    TokenInputForPoSAccountingOperation(Id<Transaction>),
+    #[error("No token outputs is allowed for PoS accounting operations {0}")]
+    TokenOutputForPoSAccountingOperation(Id<Transaction>),
```

Comment:
I would call it "TokenOutputInPoSAccountingOperation", because the presence itself is the issue, not the association.